dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java

```diff
@@ -175,6 +176,8 @@ public AbstractDirectory(URL url, RouterChain<T> routerChain, boolean isUrlFromR
                     .setProtocol(consumedProtocol)
                     .setPath(path);
             if (isUrlFromRegistry) {
+                //remove parameters related to register
+                consumerUrlFrom.removeParameter(RegistryConstants.REGISTRY_CLUSTER_KEY);
```

Comment:
Is this worked? Seems these parameters will be removed in `consumerUrlFrom.clearParameters()`